packages/core/src/providers/base.ts

```diff
@@ -1,21 +1,31 @@
 import { generateSchema } from '../generators/schema';
-import { FlatbreadConfig, LoadedFlatbreadConfig } from '../types';
+import { cache } from '../cache/cache';
+import { FlatbreadConfig } from '../types';
 import { initializeConfig } from '../utils/initializeConfig';
 
 import { graphql, GraphQLArgs, GraphQLSchema } from 'graphql';
+import { PromiseOrValue } from 'graphql/jsutils/PromiseOrValue';
 
 /**
- * Flatbread Provider
- * create a new flatbread provide that provides a query function based on a flatbread config
+ * **Flatbread Provider**
+ *
+ * Create a new Flatbread provider which contains a GraphQL `query` function that's baked with a Flatbread config
  */
 export class FlatbreadProvider {
-  private schemaPromise: Promise<GraphQLSchema>;
+  private schemaPromise: PromiseOrValue<GraphQLSchema>;
 
   constructor(config: FlatbreadConfig) {
     const initializedConfig = initializeConfig(config);
-    this.schemaPromise = generateSchema({ config: initializedConfig });
+    this.schemaPromise =
+      cache.schema ?? generateSchema({ config: initializedConfig });
```

Comment:
:fireworks:

Reply:
maybe we should move the cache check into generateSchema, so it works for all providers? and key the cache with the config in case people want multiple flatbread instances with different configs ( I have no idea why people would want that)

Reply:
ooo great idea! Adding that now

Reply:
Left a todo in there for cache busting on content category field changes. Currently it'll still get a cache hit if a user flips a `string` field to an `int`